src/Header/Authenticator.h

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef GROUP4_HEADER_AUTHENTICATOR_H_
+#define GROUP4_HEADER_AUTHENTICATOR_H_
+
+#include <string>
+
+//used to check if request is send from valid user
+class Authenticator
+{
+public:
+    //singleton code convention method.
+    unique_ptr<Authenticator> Get();
+    //check if login is correct
+    static bool CheckAccount(const std::string &account, const std::string &password);
+    //check if account exist
+    static std::string SearchAccount(const std::string &account);
+private:
+    //singleton code convention method.
+    Authenticator();
+    //non copyable
+    Authenticator(Authenticator& cpy);
```

Comment:
Please use
= delete;
to delete the method.

Reply:
Done